src/pandascore_client.py

```
@@ -0,0 +1,433 @@
+    def __init__(self, api_key: Optional[str] = None):
+        """
+        Initialize the PandaScore client.
+
+        Args:
+            api_key: PandaScore API key. If not provided, uses
+                PANDASCORE_API_KEY from config.
+        """
+        self.api_key = api_key or PANDASCORE_API_KEY
+        # Fail fast when API key is missing or empty/whitespace only.
+        if not (isinstance(self.api_key, str) and self.api_key.strip()):
+            raise ValueError("PandaScore API key is required")
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)